app/models/task.rb

```diff
@@ -110,15 +111,15 @@ def author?(user)
   end
 
   def showable_by?(user)
-    user.can?(:show, self)
+    user.nil? ? false : user.can?(:show, self)
   end
 
   def updateable_by?(user)
-    user.can?(:update, self)
+    user.nil? ? false : user.can?(:update, self)
   end
 
   def destroyable_by?(user)
-    user.can?(:destroy, self)
+    user.nil? ? false : user.can?(:destroy, self)
   end
```

Comment:
I needed showable_by? to work with user=nil and also changed the other methods for consistency.